### PR TITLE
Add support for azerty (fr) keyboard layout

### PR DESCRIPTION
**Pull request by mickael9**

### CODE REVIEW

File: lib/fr-scanedit.cpp [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+#define	SHIFT   0x80
+
+FrScanEdit::FrScanEdit() {
+    /* Generated using: loadkeys -m fr-pc | python convert.py */
+    /* https://gist.github.com/mickael9/567ce26b028ce89cafcaf4250464c526#file-convert-py */
+    keyMap = QMap<int, int>();
+    keyMap[0x09] = 0x2b; /* '\t' */
+    keyMap[0x0a] = 0x28; /* '\n' */
+    keyMap[0x1b] = 0x29; /* '\x1b' */
```
Comment (GuilloOme) [inline lib/fr-scanedit.cpp]: Since "esc" (0x1b) is not a printable char, it shouldn't be present in a password.

**Comment by klali** [comment]
I decided to not merge this straight away based on the comment from @GuilloOme 
@mickael9 if you update this pull request I'm happy to merge it.

**Comment by mickael9** [comment]
Done, and rebased on top of master.

I was also wondering if there was firmware support for other modifiers keys like AltGr?
`@` is AltGr-0 on azerty for instance.

**Comment by klali** [comment]
No, the key only supports single key presses at a time.

Thanks!
Merging this with the warning that there's no current plans for making a release of this project.

**Pull request merged**